src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -0,0 +1,358 @@
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.PlatformUI.OleComponentSupport;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Roslyn.Utilities;
+using Task = System.Threading.Tasks.Task;
+using IOleServiceProvider = Microsoft.VisualStudio.OLE.Interop.IServiceProvider;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Experimentation
+{
+    /// <summary>
+    /// Detects if keybindings have been messed up by ReSharper disable, and offers the user the ability
+    /// to reset if so.
+    /// </summary>
+    /// <remarks>
+    /// The only objects to hold permanent references to this object should be callbacks that are registered for in
+    /// <see cref="InitializeCore"/>. No other external objects should hold a reference to this. Unless the user clicks
+    /// 'Never show this again', this will persist for the life of the VS instance, and does not need to be manually disposed
+    /// in that case.
+    /// </remarks>
+    [Export(typeof(IExperiment))]
+    internal sealed class KeybindingResetDetector : ForegroundThreadAffinitizedObject, IExperiment, IOleCommandTarget, IDisposable
+    {
+        // Flight info
+        private const string InternalFlightName = "keybindgoldbarint";
+        private const string ExternalFlightName = "keybindgoldbarext";
+        private const string KeybindingsFwLink = "https://go.microsoft.com/fwlink/?linkid=864209";
+
+        // Resharper commands and package
+        private const uint ResumeId = 707;
+        private const uint SuspendId = 708;
+        private const uint ToggleSuspendId = 709;
+        private static readonly Guid ReSharperPackageGuid = new Guid("0C6E6407-13FC-4878-869A-C8B4016C57FE");
+        private static readonly Guid ReSharperCommandGroup = new Guid("{47F03277-5055-4922-899C-0F7F30D26BF1}");
+
+        private readonly VisualStudioWorkspace _workspace;
+        private readonly SVsServiceProvider _serviceProvider;
+
+        private IExperimentationService _experimentationService;
+        private IVsUIShell _uiShell;
+        private IOleCommandTarget _oleCommandTarget;
+        private OleComponent _oleComponent;
+
+        private bool _disposedValue = false;
+        private uint _priorityCommandTargetCookie = VSConstants.VSCOOKIE_NIL;
+
+        private bool _infoBarOpen = false;
+
+        [ImportingConstructor]
+        public KeybindingResetDetector(VisualStudioWorkspace workspace, SVsServiceProvider serviceProvider)
+        {
+            _workspace = workspace;
+            _serviceProvider = serviceProvider;
+        }
+
+        public Task Initialize()
+        {
+            // Immediately bail if the user has asked to never see this bar again.
+            if (_workspace.Options.GetOption(KeybindingResetOptions.NeverShowAgain))
+            {
+                return Task.CompletedTask;
+            }
+
+            return InvokeBelowInputPriority(InitializeCore);
```

Comment:
InvokeBelowInputPriority seems should be renamed to InvokeBelowInputPriorityAsync

Reply:
That's a relatively big and unrelated change. Filed https://github.com/dotnet/roslyn/issues/23661.